File: 三个数的最大乘积.cpp
#include<stdio.h>
int maximumProduct(int* nums, int numsSize) {
    int i,j,t,a,b;
    for(i=0;i<numsSize-1;i++){
        for(j=i+1;j<numsSize;j++){
            if(nums[i]<nums[j]){
                t=nums[j];
                nums[j]=nums[i];
                nums[i]=t;
            }
        }
    }
    a=nums[0]*nums[1]*nums[2];
    b=nums[0]*nums[numsSize-1]*nums[numsSize-2];
    return a>b?a:b;
}
int main(void){
	int s[5]={-4,-3,-2,-1,60};
	int t;
	t=maximumProduct(s,5);
	printf("%d\n",t);
} 

File: 2020年11月/三个数的最大乘积.py
class Solution(object):
    def maximumProduct(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums)==3:
            return nums[0]*nums[1]*nums[2]
        a=sorted(nums)
        if a[0]*a[1]*a[len(a)-1] > a[len(a)-1]*a[len(a)-2]*a[len(a)-3]:
            return a[0]*a[1]*a[len(a)-1]
        else:
            return a[len(a)-1]*a[len(a)-2]*a[len(a)-3]